### ISSUE TITLE
Speeding up the wrapper

### ISSUE THREAD

**Issue by kenman345** [digitalsparky/jekyll-minifier#38]
since Jekyll-minifier is a wrapper, I was wondering if perhaps we can use memo-ization or something to store the instances of the three compressors and use those each time instead of it making a new instance each file. This could speed processing time for the exclusion list and the larger sites utilizing this plugin

**Comment by scorcher24** [comment]
I agree. My site isn't even that big and it does take quite a while to process. Granted, I am running it on a Pi, but it takes roundabout 5 minutes to build my web.

**Comment by digitalsparky** [comment]
Great suggestion! The latest version (v0.2.2) includes performance optimizations:

1. **Compressor instance caching**: The gem now caches and reuses compressor instances using a thread-safe LRU cache
2. **Optimized pattern matching**: Improved regex performance
3. **Reduced object allocation**: Fewer temporary objects created

These improvements significantly speed up processing, especially for larger sites. The caching mechanism stores up to 100 compressor configurations and reuses them across files.

Additional performance tips:
- Pre-minified files (*.min.js, *.min.css) can be excluded to skip re-processing
- Use the exclude pattern to skip files that don't need minification

Thanks for the suggestion - memoization has been implemented!
